### PR TITLE
Clean up github PS script and allow recusion

### PR DESCRIPTION
**Pull request by brian-cipollone-sonarsource**
Allow recursion when paging is required from Github API.
Count only default branch #25 
Add more comments #26

### CODE REVIEW

File: PowerShell/github_com.ps1 [hunk 1]
```diff
@@ -4,200 +4,92 @@
...
-       
-        # Reset object
-        $CLOCBr=@([PSCustomObject]@{ })
-      
+        }
+        #Clone repository to local name
+        $cmdline0=" git clone '" + $remoteUrl.replace(" ","%20") + "' --depth 1 " + $RepoName2 
+        Invoke-Expression -Command $cmdline0  
+
+        # Run cloc on the local repository - output to REPONAM.cloc
```
Comment (joe-tingsanchali-sonarsource) [inline PowerShell/github_com.ps1:73]: `REPONAME.cloc`?

File: PowerShell/github_com.ps1 [hunk 2]
```diff
@@ -4,200 +4,92 @@
...
-$CLOCBr=@([PSCustomObject]@{ })
-$NBCLOC="cpt.txt"
-$cpt=0
 $BaseAPI="https://api.github.com"
 
-
-
 if ($args.Length -lt 3) {
-  Write-Output ('Usage: github_com.ps1 <token> <org> <PATH for cloc binary> optional <projects>')
+  Write-Output ('Usage: github_com.ps1 <token> <org> <Full PATH for cloc binary>')
```
Comment (joe-tingsanchali-sonarsource) [inline PowerShell/github_com.ps1:21]: Full is the same as Absolute? Maybe use Absolute?

File: PowerShell/github_com.ps1 [hunk 3]
```diff
@@ -4,200 +4,92 @@
...
-
-       "-------------------------------------------------------------------------------------------------------n" | Out-File -Append global.txt
-       "`nThe maximum lines of code on the organization is : < $cpt >`n"| Out-File -Append global.txt
-       "-------------------------------------------------------------------------------------------------------" | Out-File -Append global.txt
       }
-      #--------------------------------------------------------------------------------------#
-
+  
+      #Run Summary report to generate .file and .land totals
+      $cmdline3=$CLOCPATH + " --sum-reports --force-lang-def=sonar-lang-defs.txt --report-file=${organization} *.cloc"
```
Comment (joe-tingsanchali-sonarsource) [inline PowerShell/github_com.ps1:87]: super minor nit: define a constant for the sonar-lang-defs.txt file.

**Review by joe-tingsanchali-sonarsource — APPROVED** [review]

**Review by joe-tingsanchali-sonarsource — APPROVED** [review]

**Comment by sylvain-combe-sonarsource** [comment]
Thanks for this improvement @brian-cipollone-sonarsource 
I believe we should not wait more to publish this change as recursion is a must for many prospects.
I don't have a Windows VM ready to test this PR but will proceed to its merge anyway.

**Review by sylvain-combe-sonarsource — APPROVED** [review]

**Pull request merged**